docs/arch/adr-47.md

```diff
@@ -0,0 +1,69 @@
+# ADR 47: Highlighter Task
+
+1 Mar 2023
+
+## Context
+To support OCR verification (the DigiLeap project), with subjects that consist of an image and OCR output as text file, we will create a highlighter task. The highlighter task will allow volunteers to highlight text in the OCR output with a label or labels defined by researchers.
+
+The FEM highlighter task is an iteration on the [PFE highlighter task](https://github.com/zooniverse/Panoptes-Front-End/tree/master/app/classifier/tasks/highlighter).
+
+## Features
+To use the highlighter task a volunteer selects a label from the task area, highlights the related text in the text subject viewer, then selects a different label from the task area to highlight additional text in the text subject viewer or completes the task. Highlighted text can be unhighlighted.
+
+The highlighter task label(s) description and color is defined by researchers.
+
+The following is an example of a highlighter task annotation with labels "Scientific Name", "Collector Name", "Habitat", and "Location":
+```json
+{
+  "value": [
+    {
+      "labelInformation": {
+          "color": "#00FF7F",
+          "label": "Scientific Name"
+      },
+      "start": 25,
+      "end": 55,
+      "text": "Allium vineale L.(Fieldgarlic)"
+    },
+    {
+      "labelInformation": {
+          "color": "#F5D76E",
+          "label": "Collector Name"
+      },
+      "start": 402,
+      "end": 422,
+      "text": "Derick B. Poindexter"
+    },
+    {
+      "labelInformation": {
+          "color": "#ffa539",
+          "label": "Habitat"
+      },
+      "start": 207,
+      "end": 324,
+      "text": "Growing at the base of\nthe mountain, in wet meadow and adjacent trail margins Infrequent, exotic perennial\nforb/herb."
+    },
+    {
+      "labelInformation": {
+          "color": "#DCC6E0",
+          "label": "Location"
+      },
+      "start": 105,
+      "end": 206,
+      "text": "Located at Sheep Rock along Air\nBellows Gap Rd. (SR 1130) ca. 0.1 mi prior to Pruitt Rd. on the left."
+    }
+  ],
+  "task": "T0",
+  "taskType": "highlighter"
+}
+```
+
+## Decision
+Create a highlighter task in FEM.
+
+## Consequences
+Create an experimental task, the highlighter task, in FEM.
+If a workflow uses the highlighter task, add event listeners to the text subject viewer.
```

Comment:
Can the [interaction layer](https://github.com/zooniverse/front-end-monorepo/tree/master/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer) be used or is a simpler solution preferable? 

I think adding event listeners to the text subject viewer is only necessary if the user actions are select label then highlight text, and adding event listeners is not necessary if the user actions are highlight text then select label.

Reply:
Likely a bad idea, but instead of the text subject viewer as is, maybe this task would work with a text subject viewer that is a text(area) input, the value of which can't be changed, but pointer and key events to highlight and unhighlight the contents of would be easier to work with within.

Reply:
I could be wrong but isn’t the `InteractionLayer` designed to handle pointer events on interactive SVGs?

Reply:
>maybe this task would work with a text subject viewer that is a text(area) input,

A couple of questions:
- how does the current highlighter task work? I know that it was originally built to work with keyboard, touch and pointer inputs, so it would be a good starting point. 
- have you considered using `contentEditable`?

Reply:
> If the existing highlighter task meets the requirements, I’d port the code across and iterate from that starting point. That would at least get an initial prototype up in a couple of days or so.

Excellent suggestion and what I'm going to do! ✔️

Reply:
I'm having trouble using [the keyboard (Mac OS)](https://support.apple.com/en-us/HT201236) with the [PFE highlighter task](https://master.pfe-preview.zooniverse.org/projects/eatyourgreens/text-highlighter-demo/classify). My expectation was that I'd be able to tab into the text viewer and select text with arrow keys or the other Document Shortcuts linked? I'm not sure if that's right though.

Reply:
As a sighted user, I use the shortcuts after double clicking to select a word eg. Select a word then adjust the selection by a few characters if it isn’t quite right. 

You might need to ask a blind user how to select text from the keyboard with VoiceOver or NVDA. There might be instructions online.